File: 2º/PDOO/ProyectoRUBY/lib/s_ir_casilla.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative "sorpresa.rb"
require_relative "jugador.rb"

module Civitas
  class S_irCasilla<Sorpresa
    def initialize(tablero, valor, texto)
      super(texto)
      @tablero=tablero
      @valor=valor
    end
    def aplicarAJugador(actual, todos)
      super.informe(actual, todos)
      casi=todos[actual].getNumCasillaActual()
      nueva=@tabla.calcularTirada(casi, @valor)
      nueva=@tabla.nuevaPosicion(casi,nueva)
      todos[actual].moverACasilla(nueva)
      recibeJugador(actual,todos);
    end
    def to_s
      aux=super
      return aux
    end    
end
end

File: 2º/PDOO/ProyectoJAVA/src/civitas/S_irCasilla.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author nikitastetskiy
 */
public class S_irCasilla extends Sorpresa{
    
    private Tablero tablero;
    private int valor;
    
    S_irCasilla(Tablero tablero, int valor, String texto){
        super(texto);
        this.tablero = tablero;
        this.valor = valor;
    }
    
    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)) {
            informe(actual, todos);
            int casillaActual = todos.get(actual).getNumCasillaActual();
            int tirada = tablero.calcularTirada(casillaActual, valor);
            int nuevaPosicion = tablero.nuevaPosicion(casillaActual, tirada);
            todos.get(actual).moverACasilla(nuevaPosicion);
            Casilla c = tablero.getCasilla(nuevaPosicion);
            c.recibeJugador(actual, todos);
        }
    }
    
    @Override
    public String toString() {
       return super.toString();
    }
    
}
